### PR TITLE
[WIP] Updates to simplify PLT example and use new features

### PR DESCRIPTION
**Pull request by srush**
Some changes to PLT to support our workflow. Prelim support for TPUs. 

Still testing.

### CODE REVIEW

File: examples/ner/transformer_base.py [hunk 1]
```diff
@@ -103,18 +107,16 @@ def configure_optimizers(self):
...
-        self.lr_scheduler = scheduler
+        self.opt = optimizer
         return [optimizer]
 
     def optimizer_step(self, epoch, batch_idx, optimizer, optimizer_idx, second_order_closure=None):
-
-        # Step each time.
-        optimizer.step()
-        self.lr_scheduler.step()
+        if self.trainer.use_tpu:
```
Comment (williamFalcon) [inline examples/ner/transformer_base.py]: not needed any more with master.

File: examples/ner/transformer_base.py [hunk 2]
```diff
@@ -213,6 +220,7 @@ def add_generic_args(parser, root_dir):
     )
 
     parser.add_argument("--n_gpu", type=int, default=1)
+    parser.add_argument("--n_tpu", type=int, default=0)
```
Comment (williamFalcon) [inline examples/ner/transformer_base.py]: n_tpu probably better as num_tpu_cores bc 1 TPU != 1 tpu core

File: examples/ner/transformer_base.py [hunk 3]
```diff
@@ -4,6 +4,9 @@
 import numpy as np
 import pytorch_lightning as pl
 import torch
+import logging
+
+import torch_xla.core.xla_model as xm
```
Comment (williamFalcon) [inline examples/ner/transformer_base.py]: not needed anymore

File: examples/ner/transformer_base.py [hunk 4]
```diff
@@ -78,19 +84,17 @@ def __init__(self, hparams, num_labels=None):
         )
         self.config, self.tokenizer, self.model = config, tokenizer, model
         self.proc_rank = -1
+        self.is_tpu = False
+        if self.hparams.n_tpu > 0:
+            self.is_tpu = True
```
Comment (williamFalcon) [inline examples/ner/transformer_base.py]: self.trainer.on_tpu

File: examples/ner/transformer_base.py [hunk 5]
```diff
@@ -78,19 +84,17 @@ def __init__(self, hparams, num_labels=None):
...
 
     def is_logger(self):
         return self.proc_rank <= 0
```
Comment (williamFalcon) [inline examples/ner/transformer_base.py]: self.trainer.proc_rank

File: examples/ner/transformer_base.py [hunk 6]
```diff
@@ -78,19 +84,17 @@ def __init__(self, hparams, num_labels=None):
         )
         self.config, self.tokenizer, self.model = config, tokenizer, model
         self.proc_rank = -1
+        self.is_tpu = False
+        if self.hparams.n_tpu > 0:
+            self.is_tpu = True
 
     def is_logger(self):
```
Comment (williamFalcon) [inline examples/ner/transformer_base.py]: ```self.print()``` now logs from proc=0 only

File: examples/ner/transformer_base.py [hunk 7]
```diff
@@ -127,22 +129,27 @@ def test_step(self, batch, batch_nb):
     def test_end(self, outputs):
         return self.validation_end(outputs)
 
-    @pl.data_loader
     def train_dataloader(self):
-        return self.load_dataset("train", self.hparams.train_batch_size)
+        dataloader = self.load_dataset("train", self.hparams.train_batch_size)
+        t_total = (
```
Comment (williamFalcon) [inline examples/ner/transformer_base.py]: is this the small issue left?

File: examples/ner/transformer_base.py (hunk 2 above)
Comment (srush) [reply examples/ner/transformer_base.py]: Gotcha.

**Comment by BramVanroy** [comment]
I can't seem to find who is pushing the commits, but can whoever is doing that please add commit messages? This is really messy and impossible to go over as currently written.

File: examples/ner/run_pl.sh [hunk 8]
```diff
@@ -1,6 +1,20 @@
-# Require pytorch-lightning=0.6
+# Install newest ptl.
+!pip install -U git+http://github.com/PyTorchLightning/pytorch-lightning/
```
Comment (shoarora) [inline examples/ner/run_pl.sh]: is this `!` at the start supposed to be here?  or an artifact of porting from jupyter?

**Comment by srush** [comment]
Sorry, @BramVanroy . I didn't realize people were reviewing this branch. This is a WIP We are still trying to work this out with new pytorch-lightning / TPU changes. (Closing for now and will switch to a new review branch when it is working).

**Comment by BramVanroy** [comment]
@srush No worries! Wasn't really reviewing, I was just curious which changes were being made - but then I saw those commit messages and I didn't know what to expect. 😄

**Pull request closed without merging**